benchmark/analysis-scripts/autogroup.py

```diff
@@ -0,0 +1,240 @@
+    hydra_dir = os.path.join(iteration_dir, '.hydra')
+
+    # Read base config
+    config_path = os.path.join(hydra_dir, 'config.yaml')
+    if not os.path.exists(config_path):
+        return {}
+
+    # Load base config with OmegaConf
+    config = OmegaConf.load(config_path)
+
+    # Apply overrides
+    override_path = os.path.join(hydra_dir, 'overrides.yaml')
+    if os.path.exists(override_path):
+        overrides = OmegaConf.load(override_path)
+        if overrides:
+            # Convert overrides list to OmegaConf and merge
+            override_list = OmegaConf.to_container(overrides, resolve=True)
+            if isinstance(override_list, list):
+                config = OmegaConf.merge(config, OmegaConf.from_dotlist(override_list))
+
+    # Convert to regular dict and flatten
+    config_dict = OmegaConf.to_container(config, resolve=True)
+    return flatten_config(config_dict)
+
+
+def flatten_config(config: Dict[str, Any], parent_key: str = '', sep: str = '.') -> Dict[str, Any]:
+    """Flatten nested configuration dictionary."""
+    result = {}
+    for k, v in config.items():
+        new_key = f"{parent_key}{sep}{k}" if parent_key else k
+        if isinstance(v, dict):
+            result.update(flatten_config(v, new_key, sep=sep))
+        else:
+            result[new_key] = v
+    return result
+
+
+def parse_benchmark_file(file_path: str) -> Optional[float]:
+    """Parse benchmark output file and return throughput."""
+    try:
+        with open(file_path, 'r') as f:
+            data = json.load(f)
+
+        match data:
+            # CRT format
+            case {'throughput_gbps': throughput}:
+                return throughput
+
+            # Client/Prefetch format
+            case {'summary': {'total_bytes': total_bytes, 'total_elapsed_seconds': total_seconds}}:
+                return (total_bytes * 8) / (1024 * 1024 * 1024 * total_seconds)
+
+            # Client/Prefetch format with missing fields
+            case {'summary': summary}:
+                total_bytes = summary.get('total_bytes', 0)
+                total_seconds = summary.get('total_elapsed_seconds', 1)
+                return (total_bytes * 8) / (1024 * 1024 * 1024 * total_seconds)
+
+            # FIO format
+            case {'jobs': [{'read': {'io_bytes': io_bytes, 'runtime': runtime_ms}}, *_]}:
+                return (io_bytes * 8) / (1024 * 1024 * 1024 * (runtime_ms / 1000))
+
+            # FIO format with missing fields
+            case {'jobs': [job, *_]}:
+                read_data = job.get('read', {})
+                io_bytes = read_data.get('io_bytes', 0)
+                runtime_ms = read_data.get('runtime', 1000)
+                return (io_bytes * 8) / (1024 * 1024 * 1024 * (runtime_ms / 1000))
+
+            # Unknown format
+            case _:
+                print(f"{Fore.YELLOW}⚠️  Warning: Unknown format in {file_path}{Style.RESET_ALL}")
```

Comment:
Printing errors to stdout isn't ideal - should be to stderr (`print(..., file=sys.stderr)`).

Alternatively (preferred), use the [`warnings.warn`](https://docs.python.org/3/library/warnings.html#warnings.warn) function